src/sidebar/services/frame-sync.js

```diff
@@ -182,6 +182,9 @@ export default function FrameSync(annotationsService, bridge, store) {
     bridge.on('setDoodleOptions', function (state) {
       bridge.call('setDoodleOptions', state);
     });
+    bridge.on('saveCurrentDoodle', function (state) {
+      bridge.call('saveCurrentDoodle', state);
+    });
```

Comment:
These lines really make me wonder about hypothesis' internal structure, and the design decisions that led to this. Did they really need to make it so complicated with like 5 places to change?

Reply:
Feels like shotgun surgery, but they are trying to do some pretty complex stuff with the sidebar and the adder toolbar and like 3 other interfaces